lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -851,6 +851,25 @@ impl Instance {
     pub fn set_instruction_count(&mut self, instruction_count: u64) {
         self.get_instance_implicits_mut().instruction_count = instruction_count;
     }
+
+    #[inline]
+    pub fn set_hostcall_stack_reservation(&mut self, reservation: usize) {
+        let slot = self
+            .alloc
+            .slot
+            .as_ref()
+            .expect("reachable instance has a slot");
+
+        if reservation > slot.limits.stack_size {
+            panic!("TODO: error this");
```

Comment:
I'm actually not sure if making this `Result`-y _is_ what we want. It means either we make `Instance::new` a `Result`-y function, or make it panicky; a custom `Limit` the default hostcall reservation may become excessively sized.

On the other hand, someone might want to gracefully error on instances with too-small stacks?

Either way, I don't think we can push this check anywhere else (like `Limits` creation), since `set_hostcall_stack_reservation` can be called at any arbitrary point later in the `Instance`'s lifecycle.

Reply:
I ended up moving `hostcall_reservation` onto `Limits`, and added a hidden field so a `Limits` instance can only be constructed through methods that check the consistency of `stack_size`/`hostcall_reservation`. Those are now `Result`-y builder methods, and seems to be a pretty usable way to spell this out.